lib/datadog/tracing/contrib/view_component/configuration/settings.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require_relative '../../../configuration/settings'
+require_relative '../ext'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module ViewComponent
+        module Configuration
+          # Custom settings for the ViewComponent integration
+          # @public_api
+          class Settings < Contrib::Configuration::Settings
+            option :enabled do |o|
+              o.type :bool
+              o.env Ext::ENV_ENABLED
+              o.default true
+            end
+
+            # @!visibility private
+            option :analytics_enabled do |o|
+              o.type :bool
+              o.env Ext::ENV_ANALYTICS_ENABLED
+              o.default false
+            end
+
+            option :analytics_sample_rate do |o|
+              o.type :float
+              o.env Ext::ENV_ANALYTICS_SAMPLE_RATE
+              o.default 1.0
+            end
```

Comment:
Our bad here, but these are part of a really [old and deprecated feature](https://docs.datadoghq.com/tracing/legacy_app_analytics/?tab=ruby), and shouldn't be added to new code.

I don't see anywhere in your PR that uses these, so I recommend just removing them altogether. Make sure to remove any trail of it too (e.g. env vars).